### PR TITLE
Removed ReactiveUI

### PR DESCRIPTION
**Pull request by distantcam**
We're not really using ReactiveUI so I've removed it.

### CODE REVIEW

**Comment by adamralph** [comment]
👍 @distantcam I always enjoy the smell of a dependency removal 😌 

I'm afraid I need to put my 🔰 hat on and ask to be lead by the hand through these changes...

**Comment by distantcam** [comment]
@adamralph Yeah I figured. The changes are "small" in that there's only 2 things being removed (ReactiveList and ReactiveCommand) but then the replacement code is non-trivial.

We can go through it next week.

**Comment by HEskandari** [comment]
@distantcam This looks good to me. Feel free to merge after your session with Adam.

File: src/ServiceInsight/Framework/Commands/ObservableCommand.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+        public void Execute(object parameter)
+        {
+            using (StartExecuting())
+            {
+                action(parameter);
+            }
+        }
+
+        public void RaiseCanExecuteChanged()
```
Comment (adamralph) [inline src/ServiceInsight/Framework/Commands/ObservableCommand.cs]: Should this be private?

File: src/ServiceInsight/Framework/Commands/ObservableCommand.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+namespace ServiceInsight.Framework.Commands
+{
+    using System;
+    using System.Reactive.Disposables;
+    using System.Reactive.Linq;
+    using System.Threading;
+    using System.Windows.Input;
+
+    class ObservableCommand : ICommand
```
Comment (adamralph) [inline src/ServiceInsight/Framework/Commands/ObservableCommand.cs]: Does this class need to be thread-safe WRT accessing the `isExecuting` and `latest` fields? (I noticed the use `Interlocked.Exchange` in `Dispose`.)

File: src/ServiceInsight/MessageHeaders/MessageHeadersViewModel.cs [hunk 3]
```diff
@@ -29,13 +28,9 @@ public void Handle(SelectedMessageChanged @event)
...
 
-            using (KeyValues.SuppressChangeNotifications())
+            foreach (var item in headers)
             {
-                KeyValues.AddRange(headers.Select(h => new MessageHeaderKeyValue
-                {
-                    Key = h.Key,
-                    Value = h.Value
-                }));
+                KeyValues.Add(new MessageHeaderKeyValue { Key = item.Key, Value = item.Value });
```
Comment (adamralph) [inline src/ServiceInsight/MessageHeaders/MessageHeadersViewModel.cs]: Does it matter that we are no longer batching these changes up into a single change notification?

**Comment by distantcam** [comment]
@adamralph Made the changes as discussed in Lisbon. Please review.

File: src/ServiceInsight/Framework/Commands/ObservableCommand.cs [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                action(parameter);
+            }
+        }
+
+        public void RaiseCanExecuteChanged()
+        {
+            CanExecuteChanged?.Invoke(this, EventArgs.Empty);
+        }
+
+        public void Dispose()
```
Comment (adamralph) [inline src/ServiceInsight/Framework/Commands/ObservableCommand.cs]: I just realised that this class does not implement `IDisposable` and this method has no callers, so it's redundant. Should we remove it, or should this class be `IDisposable`?

File: src/ServiceInsight/Framework/Commands/ObservableCommand.cs (hunk 1 above)
Comment (adamralph) [inline src/ServiceInsight/Framework/Commands/ObservableCommand.cs]: Did we agree on this?

Comment (distantcam) [inline src/ServiceInsight/Framework/Commands/ObservableCommand.cs]: I thought we had.

File: src/ServiceInsight/MessageHeaders/MessageHeadersViewModel.cs (hunk 3 above)
Comment (adamralph) [inline src/ServiceInsight/MessageHeaders/MessageHeadersViewModel.cs]: As discussed, this is OK.

**Comment by distantcam** [comment]
@adamralph Please review

**Pull request merged**